File: docs/examples/python/simple_function.py
#!/usr/bin/env python3


def add_prefix(function):
    def dec(*args, **kwargs):
        print('Print message:')
        print(function(*args, **kwargs))

    return dec


@add_prefix
def some_user_func():
    return 'Hello world'


if __name__ == '__main__':
    some_user_func()

File: docs/examples/php/simple_function.php
#!/usr/bin/env php
<?php

require_once __DIR__ . '/autoload.php';

#[Attribute(Attribute::TARGET_FUNCTION | Attribute::IS_REPEATABLE)]
readonly final class add_prefix
{
    public function __invoke(callable $function): callable
    {
        echo 'Print message:' . PHP_EOL;
        echo $function();

        return function (): void {
        };
    }
}

#[add_prefix]
function simple_function(): string
{
    return 'Hello world' . PHP_EOL;
}

call_user_func_with_decorators('simple_function');
